deployments/cognito/terraform/cognito-components/main.tf

```diff
@@ -0,0 +1,270 @@
+provider "aws" {
+  alias = "aws"
+}
+
+provider "aws" {
+  alias = "virginia"
```

Comment:
what about region? alias just a name

Reply:
this is an alias for a provider that is passed into the module, so the provider that is passed in must declare the region (unless that provider itself was an alias)

Reply:
specifically in the root module main.tf there is:
```
provider "aws" {
  region = local.region
}

# Cognito requires a certificate in N.Virginia in order to have a custom domain for a user pool
# https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-add-custom-domain.html
provider "aws" {
  region = "us-east-1"
  alias = "virginia"
}
```

which is wired down to the dependent modules:
```
module "kubeflow_components" {
  source = "./cognito-components"

  ...
  
  providers = {
    aws = aws
    aws.virginia = aws.virginia
  }
}
```